gui/wxpython/tplot/frame.py

```diff
@@ -46,7 +46,7 @@
             'The Temporal Plot Tool needs the "matplotlib" '
             "(python-matplotlib) package to be installed. {0}"
         ).format(e)
-    )
+    ) from e
```

Comment:
I tried to understand the issue. Not sure if I got it completly right. My interpretation would be, that authors did not want excepting-chaining when they included the previous exception text with `format` in the raised exception...
So `from e` would print _e_ twice in this case, no? So probably, `from None` is the right choice when the original Error message is included in the raised exception? Not sure when one should raise a python error instead of a eg. _gs.fatal(_())_ ... Maybe @wenzeslaus has some thoughts on exception chaining?

Reply:
I don't think exception chaining was used at all before. We're way more precise about our python usage now than when the python code started, so I'm pretty sure not everything was thoroughly though out before.

But for this case, maybe forgetting the `.format(e)` and using exception chaining for the details is in the same spririt?
The spirit (as I understand), was to have better error messages that what python was giving at the time. Now, especially in the latests versions, have been giving better error messages, so it's less frightening and more useful. I think there's value of having the real errors showed up (as we will be having that in the logs in the issues the users will file), and we might want to have all that extra context.

Reply:
Oh, it's only a hint for an import error. So that's kinda easy and has no real implications for the logic of the code itself. It's only a nice-to-have hint.

So, I'm suggesting exception chaining without the format(e) here, it will give something similar to the example of python docs:
```pycon
>>> def func():
...    raise ConnectionError
... 
>>> try:
...     func()
... except ConnectionError as exc:
...     raise RuntimeError('Failed to open database') from exc
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "<stdin>", line 2, in func
ConnectionError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
RuntimeError: Failed to open database
``` 

```pycon
>>> def func():
...    raise ConnectionError
... 
>>> try:
...     func()
... except ConnectionError as exc:
...     raise RuntimeError('Failed to open database') from exc
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "<stdin>", line 2, in func
ConnectionError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
RuntimeError: Failed to open database
```


```python
try:
    import matplotlib as mpl
    # The recommended way to use wx with mpl is with the WXAgg
    # backend.
    mpl.use("WXAgg")
    from matplotlib.figure import Figure
    from matplotlib.backends.backend_wxagg import (
        FigureCanvasWxAgg as FigCanvas,
        NavigationToolbar2WxAgg as NavigationToolbar,
    )
    import matplotlib.dates as mdates
except ImportError as e:
    raise ImportError(
        _(
            'The Temporal Plot Tool needs the "matplotlib" '
            "(python-matplotlib) package to be installed."
        )
    ) from e
```

giving (except for the line numbers and file names, I wrote it by hand):
```pycon
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "<stdin>", line 2, in func
ImportError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
ImportError: The Temporal Plot Tool needs the "matplotlib" (python-matplotlib) package to be installed.
```

Reply:
Or maybe we'd want to use exception notes: https://docs.python.org/3/tutorial/errors.html#enriching-exceptions-with-notes?

Reply:
Both exception chaining and exception notes seem OK to me. However, I tend to think that all tracebacks also could be considered bugs, and when we catch an exception with try except, a simple (error-) message should be given that is addressed to the user and not to a developer who should be able to read traceback...
Maybe this discussion would make a good startingpoint to improve the exception-handling section in the style-guide?!?

Reply:
For import errors, it seems weird to me that we use dependencies, but don't list them or require them as any other python package. But that's a bit because we don't have a real python package neither, and it brings these kinds of problems.

Reply:
We shouldn't be trying to have nicer errors for import errors, if the requirements should've been installed normally and automatically as a dependency). It isn't the case right now, so it's fine to have these.

I think I'll add a TODO in the changed code. But I'll use exception chaining to align with the rest of the PR. Exception notes might be a good idea for all other cases otherwhere, but not in this.

Reply:
As for dependencies, see also https://trac.osgeo.org/grass/ticket/2895

Reply:
Even then, that ticket is starting to get old, now everything should be in a pyproject.toml. For addons, I think each should be an individual "real" python package, each with a pyproject.toml, and could be fetched only for that subdirectory and work perfectly (and allow code tools to understand that structure).

Reply:
Making addons python packages is also the way QGIS is going, I think. However, we also have C-addons and addons using R...

Reply:
I didn't go with the TODO finally. Still working on the next PR, the unsafe fixes are a little longer to go through, and I have to do more manual tuning.

Reply:
@neteler do you have any opions/thoughts on raised exceptions, exception chaining vs. GRASS messages (fatal/error)?
See also my comment here: https://github.com/OSGeo/grass/pull/4032#discussion_r1676829971